src/pip/_internal/index.py

```diff
@@ -849,25 +865,20 @@ def _get_content_type(url, session):
 
         return resp.headers.get("Content-Type", "")
 
-    @cached_property
-    def base_url(self):
-        bases = [
-            x for x in self.parsed.findall(".//base")
-            if x.get("href") is not None
-        ]
-        if bases and bases[0].get("href"):
-            return bases[0].get("href")
-        else:
-            return self.url
-
     @property
     def links(self):
```

Comment:
I see the responsibility of the property is being expanded also to parse. Since normally, properties should "do" very little, you might want to add a comment saying the property should only be called once. (I'm assuming this is just a temporary intermediate step, so I'm okay with leaving it with an expanded scope.)

Reply:
I totally agree. This property is only left as-is to reduce unnecessary diff, since the property is going to be removed in the next patch.

Reply:
On second thought, I’m going to rename the property (and turn it into a generator function) to clarify its purpose. I hate properties anyway, personally.